_ingest-pipelines/processors/json.md

```diff
@@ -0,0 +1,200 @@
+---
+layout: default
+title: JSON
+parent: Ingest processors
+nav_order: 170
+---
+
+# JSON processor
+
+The `json` processor serializes a string-valued field into a map of maps, which can be useful for various data processing and enrichment tasks.
+
+The following is the syntax for the `json` processor:
+
+```json
+{
+  "processor": {
+    "json": {
+      "field": "<field_name>",
+      "target_field": "<target_field_name>",
+      "add_to_root": <boolean>
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+## Configuration parameters
+
+The following table lists the required and optional parameters for the `json` processor.
+
+Parameter | Required/Optional | Description |
+|-----------|-----------|-----------|
+`field` | Required | The name of the field containing the JSON-formatted string to be deserialized.
+`target_field` | Optional | The name of the field where the deserialized JSON data will be stored. If this parameter is not provided, the deserialized data is added to the root of the document.
+`add_to_root` | Optional | A boolean flag that determines whether the deserialized JSON data should be added to the root of the document (`true`) or stored in the target_field (`false`). Default value is `false`. 
```

Comment:
`add_to_root` and `target_field` are partly exclusive, if `add_to_root` is `true`, `target_field` cannot be set.